walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,97 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Base cases that should return default results
+    // if there is no attendee, return the whole day as available
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalGuests = request.getOptionalAttendees();
+    if (attendees.isEmpty() && optionalGuests.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    // if the meeting request duration is longer than a day return no option
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    List<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+    // schedule for one attendee, the person initially starts free all day
+    meetingTimes.add(TimeRange.WHOLE_DAY);
+    // go through every event, mark each event the person's involved with 
+    for (Event event : events) {
```

Comment:
If you were to convert everything and use RangeSet using minutes, this code (ignoring optional attendees) could look like:

```java
RangeSet<Integer> possibleTimes = TreeRangeSet.create();
possibleTimes.add(Range.closedOpen(0, 24 * 60));
possibleTimes.removeAll(
    events.stream()
        .filter(event -> !Collections.disjoint(event.getAttendees(), attendees))
        .map(FindMeetingQuery::helperMethodConvertingEventToRange)
        .collect(ImmutableRangeSet.toRangeSet()));
return possibleTimes.asRanges()
    .stream()
    .map(FindMeetingQuery::helperMethodConvertingRangeToTimeRange)
    .collect(Collectors.toList());
```